### PR TITLE
feat: add new methods for server variable support

### PR DESCRIPTION
**Pull request by domharrington**
## 🧰 What's being changed?

Oas class changes to support editable server urls component

## 🧪 Testing

See unit tests

### CODE REVIEW

File: __tests__/tooling/index.test.js [hunk 1]
```diff
@@ -42,6 +42,138 @@ describe('#url()', () => {
...
+      new Oas({
+        servers: [{ url: 'https://example.com/{path}', variables: { path: { enum: ['v1', 'v2'] } } }],
+      }).splitUrl()[1].enum
+    ).toStrictEqual(['v1', 'v2']);
+  });
+});
+
+describe('#variables([selected])', () => {
+  it('should return with list of variables', () => {
+    const variables = { path: { description: 'path description' } };
```
Comment (erunion) [inline __tests__/tooling/index.test.js:115]: Can you add `type: int` to these variable examples so they look more real?

**Review by erunion — APPROVED** [review]
Minor test comment but otherwise looks good! I can get you a build first thing Monday morning.

**Review by Dashron — APPROVED** [review]

File: __tests__/tooling/index.test.js (hunk 1 above)
Comment (domharrington) [reply __tests__/tooling/index.test.js:115]: I dont think they have types? https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#serverVariableObject Unless i'm misunderstanding here.

Comment (erunion) [reply __tests__/tooling/index.test.js:115]: Ah no you're right, I was looking at this on mobile and was thinking I was looking at path params, not path vars.

**Comment by erunion** [comment]
@domharrington Published to 10.3.0

**Pull request merged**